cmake/common.cmake

```diff
@@ -447,6 +447,126 @@ sign_app_library(
 )
 # SNIPPET_END: JS generic application
 
+option(ENABLE_V8 "Enable building of the js_v8 app" OFF)
+if(ENABLE_V8)
+  message(STATUS "WARNING: V8 utilisation is experimental")
+
+  option(V8_DEBUG "Use V8 debug build" OFF)
+  if(V8_DEBUG)
+    set(V8_BUILD_TYPE "debug")
+  else()
+    set(V8_BUILD_TYPE "release")
+  endif()
+  message(STATUS "Using V8 ${V8_BUILD_TYPE} build")
+
+  set(v8_include_dir_relative include)
+  set(v8_lib_relative lib/libv8_monolith.a)
+  set(v8_base_dir ${CMAKE_CURRENT_SOURCE_DIR}/build-v8)
+  set(v8_virtual_dir ${v8_base_dir}/${V8_BUILD_TYPE}-virtual)
+  set(v8_virtual_include_dir ${v8_virtual_dir}/${v8_include_dir_relative})
+  set(v8_virtual_lib ${v8_virtual_dir}/${v8_lib_relative})
+  set(v8_sgx_dir ${v8_base_dir}/${V8_BUILD_TYPE}-sgx)
+  set(v8_sgx_include_dir ${v8_sgx_dir}/${v8_include_dir_relative})
+  set(v8_sgx_lib ${v8_sgx_dir}/${v8_lib_relative})
+
+  set(v8_defs V8_CC_MSVC=0)
+
+  set(js_v8_dir ${CCF_DIR}/src/apps/js_v8)
+  set(js_v8_src
+      ${js_v8_dir}/js_v8_base.cpp
+      ${js_v8_dir}/v8_runner.cpp
+      ${js_v8_dir}/v8_util.cpp
+      ${js_v8_dir}/kv_module_loader.cpp
+      ${js_v8_dir}/tmpl/console_global.cpp
+      ${js_v8_dir}/tmpl/ccf_global.cpp
+      ${js_v8_dir}/tmpl/request.cpp
+      ${js_v8_dir}/tmpl/request_authn_identity.cpp
+      ${js_v8_dir}/tmpl/request_body.cpp
+      ${js_v8_dir}/tmpl/string_map.cpp
+      ${js_v8_dir}/tmpl/kv_store.cpp
+      ${js_v8_dir}/tmpl/kv_map.cpp
+      ${js_v8_dir}/tmpl/historical_state.cpp
+      ${js_v8_dir}/tmpl/receipt.cpp
+      ${js_v8_dir}/tmpl/consensus.cpp
+      ${js_v8_dir}/tmpl/historical.cpp
+  )
+
+  if("virtual" IN_LIST COMPILE_TARGETS)
+    add_library(js_v8_base.virtual STATIC ${js_v8_src})
+    add_san(js_v8_base.virtual)
+    add_warning_checks(js_v8_base.virtual)
+    target_include_directories(
+      js_v8_base.virtual PRIVATE ${js_v8_dir} ${v8_virtual_include_dir}
+    )
+    target_link_libraries(
+      js_v8_base.virtual PUBLIC ccf.virtual ${v8_virtual_lib}
+    )
+    target_compile_options(js_v8_base.virtual PRIVATE ${COMPILE_LIBCXX})
+    target_compile_definitions(
+      js_v8_base.virtual PUBLIC INSIDE_ENCLAVE VIRTUAL_ENCLAVE
+                                _LIBCPP_HAS_THREAD_API_PTHREAD ${v8_defs}
+    )
+    set_property(
+      TARGET js_v8_base.virtual PROPERTY POSITION_INDEPENDENT_CODE ON
+    )
+    use_client_mbedtls(js_v8_base.virtual)
+    install(
+      TARGETS js_v8_base.virtual
+      EXPORT ccf
+      DESTINATION lib
+    )
+  endif()
+
+  option(ENABLE_V8_SGX
```

Comment:
This flag will go away, either once the SGX build is working or when the SGX build is abandoned. The purpose of the flag is to opt-in to the experimental failing SGX build. The currently "officially" supported target for v8 is only "virtual", and I did some minor extensions to CMake to handle the case where an app doesn't support all compile targets. 
Yes, the semantics of `COMPILE_TARGETS` are now "for each library, build the intersection of COMPILE_TARGETS and whatever targets the library supports".
This may not be the best solution, feel free to suggest something better.

Reply:
Why can't the single ENABLE_V8 flag enable the building of v8 for all COMPILE_TARGETS as appropriate? I think it's fine to mark it as experimental in all cases, I understand the sgx variant is much less functional than virtual, but neither of them is released/supported at this point.

Reply:
If we do that then we need a good way to exclude the sgx variant from tests. I haven't found one that actually works well.

Reply:
As discussed, and given that SGX seems to be working, I removed all this again.